### PR TITLE
fix(function call proposal): fix create and vote for fc proposal

### PR DESCRIPTION
**Pull request by VERZUOL1**
now gas field is editable and can be updated by user

### CODE REVIEW

File: astro_2.0/components/TransactionDetailsWidget/TransactionDetailsWidget.module.scss [hunk 1]
```diff
@@ -43,15 +43,20 @@
...
 
 .transactionDetails {
+  align-items: flex-end;
   display: flex;
   flex-basis: 42%;
-  justify-content: space-evenly;
+  justify-content: flex-start;
   width: 100%;
+
+  > div {
```
Comment (wnwd-oleksii-p) [inline astro_2.0/components/TransactionDetailsWidget/TransactionDetailsWidget.module.scss:57]: We should avoid tightening of layout with styles. Usage of tags should be avoided if can be done by styles.

File: astro_2.0/components/TransactionDetailsWidget/TransactionDetailsWidget.module.scss [hunk 2]
```diff
@@ -74,13 +79,65 @@
...
     flex-direction: column;
   }
 
   .transactionDetails {
-    justify-content: center;
+    justify-content: space-between;
+    margin-bottom: 12px;
+    padding: 0 24px;
+
+    > div {
```
Comment (wnwd-oleksii-p) [inline astro_2.0/components/TransactionDetailsWidget/TransactionDetailsWidget.module.scss:138]: Same here

File: astro_2.0/features/CreateProposal/helpers/rootHelper.tsx [hunk 3]
```diff
@@ -804,6 +817,13 @@ export function getValidationSchema(
           ),
         details: yup.string().required('Required'),
         externalUrl: yup.string().url(),
+        gas: yup
+          .number()
+          .typeError('Must be a valid number.')
+          .positive()
+          .min(0.01)
+          .max(0.25)
+          .required('Required'),
```
Comment (wnwd-oleksii-p) [inline astro_2.0/features/CreateProposal/helpers/rootHelper.tsx:826]: Maybe, there is sense to place this validation in a variable and reuse it everywhere?

File: astro_2.0/components/TransactionDetailsWidget/TransactionDetailsWidget.module.scss (hunk 1 above)
Comment (AlexeyPonomarenko) [inline astro_2.0/components/TransactionDetailsWidget/TransactionDetailsWidget.module.scss]: We should avoid tightening of layout with styles. Usage of tags should be avoided if can be done by styles.

File: astro_2.0/components/TransactionDetailsWidget/TransactionDetailsWidget.module.scss (hunk 2 above)
Comment (AlexeyPonomarenko) [inline astro_2.0/components/TransactionDetailsWidget/TransactionDetailsWidget.module.scss]: Same here

File: astro_2.0/features/CreateProposal/helpers/rootHelper.tsx (hunk 3 above)
Comment (AlexeyPonomarenko) [inline astro_2.0/features/CreateProposal/helpers/rootHelper.tsx]: Maybe, there is sense to place this validation in a variable and reuse it everywhere?

File: astro_2.0/components/TransactionDetailsWidget/TransactionDetailsWidget.module.scss (hunk 1 above)
Comment (VERZUOL1) [reply astro_2.0/components/TransactionDetailsWidget/TransactionDetailsWidget.module.scss]: fixed

File: astro_2.0/components/TransactionDetailsWidget/TransactionDetailsWidget.module.scss (hunk 2 above)
Comment (VERZUOL1) [reply astro_2.0/components/TransactionDetailsWidget/TransactionDetailsWidget.module.scss]: fixed

File: astro_2.0/features/CreateProposal/helpers/rootHelper.tsx (hunk 3 above)
Comment (VERZUOL1) [reply astro_2.0/features/CreateProposal/helpers/rootHelper.tsx]: fixed

**Review by AlexeyPonomarenko — DISMISSED** [review]

**Review by AlexeyPonomarenko — APPROVED** [review]

**Pull request merged**